ampd/src/health_check.rs

```diff
@@ -0,0 +1,105 @@
+use error_stack::{Result, ResultExt};
+use std::net::SocketAddr;
+use thiserror::Error;
+use tracing::info;
+
+use axum::{http::StatusCode, routing::get, Json, Router};
+use serde::{Deserialize, Serialize};
+use tokio_util::sync::CancellationToken;
+
+pub struct Server {
+    listener: tokio::net::TcpListener,
+}
+
+#[derive(Error, Debug)]
+pub enum HealthCheckError {
+    #[error("Health check server error: {0}")]
+    Error(String),
+}
+
+impl Server {
+    pub async fn new(bind_addr: SocketAddr) -> Result<Self, HealthCheckError> {
+        Ok(Self {
+            listener: tokio::net::TcpListener::bind(bind_addr)
```

Comment:
Normally i do not like to use `new()` for doing things like binding sockets. Initially i consider this approach doable taking into account how the rest of the codebase interacts with this server.

Reply:
Could you share why you think `new` may not be ideal for this case?

Reply:
@tilacog In this scenario, I typically reserve the use of new exclusively for configuring structs, abstaining from creating any resources. Instead, within the run function, I manage all resource acquisition. This approach fosters clearer responsibility segregation and enhances intuition. I believe reserving the socket allocation in the operating system is more suitable for the run function.

Reply:
I agree with @eloylp, it would be preferable to just store the address in the `new` function and then start the listener in the `run` function, so there is only one place that can return errors

Reply:
@cgorenflo thanks for your review. The following actions were performed for this PR:

1. Applied all your suggestions. 
2. Move the listener creation to the `run` function at 571eeaffb06c41b084de109819008ec7a7d92517 .
3. Clippy checks.
4. Execute both, unit tests and [manual tests](https://github.com/axelarnetwork/axelar-amplifier/pull/344#issuecomment-2049722752) again.

We hope these changes are the expected ones. Let us know if there's anything else you'd like us to tweak or if we are good to go.